bite-size-notes/week-2-d-Intro-To-Convex-Optimization.md

```diff
@@ -0,0 +1,143 @@
+* Norm balls and norm cones  
+* Polyhedra  
+* Positive semidefinite cone
+
+For more information on the above convex sets, take a look at [1]
+
+**Convex Functions:**  
+* $f: \pmb{R_n} \rightarrow \pmb{R}$ is convex if the domain of $f$ is a convex set and for all $x,y \in f, 0 \leq \theta\leq1$
+  
+<p align="center">
+  <img src="figs/ConvexFunc.jpg" alt="drawing" width="500"/>
+</p>   
+[1]  
+
+* $f$ is concave if -f is convex
+* $f$ is strictly convex if the domain of $f$ is convex and for $x,y \in$ domain of $f, x \neq, 0 <\theta< 1,$ 
+
+Example of convex functions: 
+
+* affine: ax+b on **R** for any $a,b \in \pmb{R}$
+* exponential: $e^{ax}$ for any $a \in \pmb{R}$
+* powers: $x^a$ on $\pmb{R_{++}}$ for $\alpha \geq 1,  \alpha \leq 0$
+* powers of absolute value: $|x|^p$ on $\pmb{R}$ for $p \geq 1$
+* positive part (relu): max {0,x} 
+
+## Main Body 
+
+### Standard form of an optimization problem 
+minimize x  in  $$f_0 (x)$$    
+   
+subject to  $$f_i (x) \leq 0, i = 1, ...., m$$ 
+             $$g_i (x) = 0, i = 1, ...., p$$
+* $x \in \pmb{R^n}$ is (vector) variable to be chosen (n scaler variables $x_1, ....., x_n$)
+* $f_0$ is the **objective function** to be minimized
+* $f_1, ..., f_m$ are the **inequality constraint functions**
+* $g_1, ..., g_p$ are the **equality constraint functions**  
+[1]
+
+### Standard form of convex optimization problem
+minimize x in  $$f_0 (x)$$    
+ 
+subject to  $$f_i (x) \leq 0, i = 1, ...., m$$ 
+             $$Ax = b$$
+* $x \in \pmb{R^n}$
+* equality constraints are linear
+* $f_1, ... f_m$ are **Convex**: for $\theta \in$ [0,1],  
+$$f_i(\theta x+ (1- \theta)y) \leq \theta f_i(x) + (1-\theta)f_i(y))$$  
+i.e., $f_i$ have nonnegative (upward) curvature
+[1]
+
+The main difference that convex optimization introduces is that both the objective function and the feasible set must now be convex. [2] 
+
+Two types of problems that we may come across in controls are: 
+
+#### Linear Programming  
+
+Simply put, if an optimization problem's requirements and objectives are linear relationships, then it is called linear programming.  
+
+Standard form  
+
+* **Linear (or Affine) function to be minimized**
+  eg. $$f(x_1, x_2) = c_1x_1 + c_2x_2$$
+* **Problem constraints**
+  eg. $$a_{11}x_1 + a_{12}x_2 \leq b_1$$
+   $$a_{21}x_1 + a_{22}x_2 \leq b_2$$
+* **non-negative variable**
+  eg. $$x_1 \geq 0$$
+  $$x_2 \geq 0$$ [4]
+
+#### Quadratic Programming  
+For quadratic programming problems, the objective function is a multivariate quadratic subject to linear constraints on its variables. 
+
+Standard form  
+
+* **Quadratic Convex function**
+  $$f_0(x) = c^Tx + x^TQx$$
+  for some vector $c \in \pmb{R}^n$ and $Q = Q^T \geq 0$
+  - Q is positive semidefinite: it is symmetric, and all eigenvalues are not negative. 
+  
+* **Problem constraints**
+  $$a_{1}x  \leq b_1$$
+  $$a_{2}x \leq b_2$$
+
+**note:**  
+Linear programming is a special case of quadratic programming where Q is zero [5]
+
+
+
```

Comment:
You could include an image of the optimization landscape of a convex optimization problem, especially with constraints, to show that the convex function has one global minimum and how constraints may change where optimal solution is